src/Instrument/Transformer/FilterInjectorTransformer.php

```diff
@@ -122,66 +124,32 @@ public static function rewrite($originalResource, $originalDir = '')
      */
     public function transform(StreamMetaData $metadata)
     {
-        if ((strpos($metadata->source, 'include') === false) && (strpos($metadata->source, 'require') === false)) {
+        $includeExpressionFinder = new NodeFinderVisitor([Include_::class]);
+
+        // TODO: move this logic into walkSyntaxTree(Visitor $nodeVistor) method
+        $traverser = new NodeTraverser();
```

Comment:
Have you think about to maybe have:

    $nodes = NodeFinderVisitor::getInstance()->findNodes([]);

? `NodeFinderVisitor` could also be a `final` class and annotated as internal...

While we are at final classes - you should start to think about making majority of classes final and annotated as internal - it will be much easier to introduce changes without BCs.

Reply:
I like the idea of making classes final and providing API for public part. Likely, AOP is low coupled and provided only small API via `Joinpoint` object instances, all this classes can already tagged via `@api` hash tag and part of them are final.

What about annotating methods as internal: in IDE all methods will be displayed striked like this ~~->getFoundNodes();~~ and this will add false sense that methods are deprecated.

I have a willing to rewrite transformers for 3.0 and break BC for that. What I want is to use `goaop/ast-transformer` library that will contain simple node visitors (`PhpParser\NodeVisitor` instances) and instead of scanning nodes in each transformer, just do this in one cycle. `goaop/ast-transformer` should also provide API for modification of original source code, hiding all details about token positions, updating of AST during transformation, etc...

Reply:
👍